src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs

```diff
@@ -40,20 +40,309 @@ public async Task GetInvitationsByCommPkgNo_AsViewer_ShouldGetInvitations()
+
+            var completerPerson = invitation.Participants
+                .Single(p => p.Organization == Organization.Contractor).Person;
+
+            if (completerPerson.Person.Id != null)
+            {
+                var completePunchOutDto = new CompletePunchOutDto()
+                {
+                    InvitationRowVersion = invitation.RowVersion,
+                    ParticipantRowVersion = completerPerson.Person.RowVersion,
+                    Participants = new List<ParticipantToChangeDto>
+                    {
+                        new ParticipantToChangeDto
+                        {
+                            Id = (int)completerPerson.Person.Id,
+                            Note = "Some note about the punch round or attendee",
+                            RowVersion = completerPerson.Person.RowVersion,
+                            Attended = true
+                        }
+                    }
+                };
+
+            // Act
+            await InvitationsControllerTestsHelper.CompletePunchOutAsync(
+                UserType.Completer,
+                TestFactory.PlantWithAccess,
+                invitationToCompleteId,
+                completePunchOutDto);
+            }
+
+            // Assert
+            var completedInvitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Viewer,
+                TestFactory.PlantWithAccess,
+                invitationToCompleteId);
+
+            var completingParticipant =
+                completedInvitation.Participants.Single(p => p.Person?.Person.Id == completerPerson.Person.Id);
+            Assert.AreEqual(IpoStatus.Completed, completedInvitation.Status);
+            Assert.IsNotNull(completingParticipant.SignedAtUtc);
+            Assert.AreEqual(_conradContractor.UserName, completingParticipant.SignedBy);
+        }
+
+        [TestMethod]
+        public async Task AcceptPunchOut_AsAccepter_ShouldAcceptPunchOut()
+        {
+            // Arrange
+            var invitationToAcceptId = await InvitationsControllerTestsHelper.CreateInvitationAsync(
+                UserType.Planner,
+                TestFactory.PlantWithAccess,
+                "InvitationForAcceptingTitle",
+                "InvitationForAcceptingDescription",
+                InvitationLocation,
+                DisciplineType.DP,
+                _invitationStartTime,
+                _invitationEndTime,
+                _participantsForSigning,
+                _mcPkgScope,
+                null
+            );
+
+            var invitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Viewer,
+                TestFactory.PlantWithAccess,
+                invitationToAcceptId);
+
+            var completerPerson = invitation.Participants
+                .Single(p => p.Organization == Organization.Contractor).Person;
+
+            if (completerPerson.Person.Id != null)
```

Comment:
Hmm ... in this case you could get a green test caused by Person.Id == null. Probably not a good test
Remove if since Person.Id is an int